module-system/module-implementations/sov-bank/src/call.rs

```diff
@@ -93,6 +107,34 @@ impl<C: sov_modules_api::Context> Bank<C> {
 
         Ok(burn_response)
     }
+
+    pub(crate) fn mint(
+        &self,
+        coins: Coins<C>,
+        minter_address: C::Address,
+        context: &C,
+        working_set: &mut WorkingSet<C::Storage>,
+    ) -> Result<CallResponse> {
+        let mut token = self.tokens.get_or_err(&coins.token_address, working_set)?;
+        let mint_response = token.mint(context.sender(), &minter_address, coins.amount, working_set)?;
+        token.total_supply += coins.amount;
```

Comment:
Let's make the `mint` function to change the `token.total_supply ` this way all the minting logic  is in one place.

Reply:
sounds good !